editor/src/ast/ast.rs

```diff
@@ -95,9 +96,20 @@ impl<'l> Ast<'l> {
         &self.tree.data().notation
     }
 
-    /// Replace this node's `i`th child. Return the replaced child. If `tree`
-    /// cannot be inserted because it has the wrong Sort, return it as
-    /// `Err(tree)`.
+    /// Create a new `hole` node that belongs to the same forest as this node.
+    pub fn new_hole(&mut self) -> Ast<'l> {
+        let node = Node {
+            bounds: Bounds::empty(),
+            language: self.get_language(),
+            construct: Construct::hole(),
+            notation: NotationSet::hole(),
+        };
```

Comment:
This (`new_hole()`) fixes the notation of all holes, making it impossible to ever change how holes are displayed. I don't know how to fix this... we need to talk about it. Open an issue?